src/GoogleAuthenticator/Helper.php

```diff
@@ -92,9 +92,15 @@ public function generateSecret()
     /**
      * @return string
      */
-    public function getSessionKey(UsernamePasswordToken $token)
+    public function getSessionKey(TokenInterface $token)
     {
-        return sprintf('sonata_user_google_authenticator_%s_%s', $token->getProviderKey(), $token->getUsername());
+        if (method_exists($token, 'getProviderKey')) {
```

Comment:
Please check for the instance instead of method existence

Reply:
That narrows the scope (couldn't there be other token types than UsernamePasswordToken that contain getProviderKey?), but I'm OK with it. I have made the change but FlintCI is now failing on some unrelated files.